mdformat/_api.py

```diff
@@ -1,17 +1,22 @@
 from pathlib import Path
-from typing import Union
+from typing import Iterable, Union
 
 from markdown_it import MarkdownIt
 
 from mdformat._renderer import MDRenderer
+import mdformat.plugins
 
 
-def text(md: str) -> str:
+def text(md: str, *, codeformatters: Iterable[str] = ()) -> str:
     """Format a Markdown string."""
-    return MarkdownIt(renderer_cls=MDRenderer).render(md)
+    markdown_it = MarkdownIt(renderer_cls=MDRenderer)
+    markdown_it.options["codeformatters"] = {
+        lang: mdformat.plugins.CODEFORMATTERS[lang] for lang in codeformatters
+    }
```

Comment:
@chrisjsewell I want to pass config to `MDRenderer` here. More specifically I need to give it a list of coding languages that I want to enable formatting for.

Question is, is this the way way to pass config to the renderer, or is there a more idiomatic way? Other ways I could think of were
- `MDRenderer.__init__` (but markdown-it-py wont let me change the args)
- Config arg of `MarkdownIt` (but that overrides all config, I just want to add this one dict in)
- `MarkdownIt.configure()` and `MarkdownIt.configure()` (i guess the same override problem here?)

Reply:
Yes I would say that this is the correct way, and how I do it in myst-parser 👍